zend_abstract_interface/hook/hook.c

```diff
@@ -226,6 +231,16 @@ static void zai_hook_hash_destroy(zval *zv) {
     efree(hooks);
 }
 
+static inline bool zai_hook_is_excluded(zai_hook_t *hook, zend_class_entry *ce) {
+    if (hook->exclusions.arData && ce) {
+        zend_string *lower = zend_string_tolower(ce->name);
+        bool excluded = zend_hash_exists(&hook->exclusions, lower);
```

Comment:
That means we only strictly check for the equality of names, which means a class extending `NullLogger` would also get hooked.

I think we do have to iterate through the interfaces `ce->interfaces` + the parents as well

Reply:
In the .phpt, I think that if we add a class `GrandChild` inherits `Child`, it would get hooked

Reply:
Okay, it's a bit more complicated with interfaces: Every class implements all the interfaces any of its parents implement.

The underlying problem is, given a hook on X::foo():
```
interface X { function foo(); }
class A implements X {  function foo() { ... } }
class B extends A { function foo() { ... } }
class C extends B implements X {}
```

I think we agree that if a class (here: B) is directly implementing A, it probably shouldn't be hooked if A::foo() is dropped.
However, now C implements X directly. Should or should it not inherit? The catch though here is that PHP does not expose whether the interface is inherited from the parent only or directly inherited by the child.

I think it's fine to ignore "implements directly".